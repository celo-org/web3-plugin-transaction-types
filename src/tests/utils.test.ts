import { describe, expect, test } from "bun:test";
import {
  CeloChains,
  getContractAddress,
  isCel2,
  isWhitelisted,
  TxTypeToPrefix,
} from "../utils";
import Web3 from "web3";
import { CeloTransactionTypesPlugin } from "..";
describe("CeloChains", () => {
  test("chains should be configured", () => {
    expect(CeloChains).toMatchSnapshot();
  });
});

describe("Types", () => {
  test("tx types should be configured", () => {
    expect(TxTypeToPrefix).toMatchSnapshot();
  });
});

describe("getContractAddress()", () => {
  const web3 = new Web3(CeloChains.alfajores.rpcUrl);
  const plugin = new CeloTransactionTypesPlugin();
  web3.registerPlugin(plugin);
  test("returns a contract address", async () => {
    expect(getContractAddress(plugin, "FeeCurrencyWhitelist")).resolves.toMatch(
      /0x[0-9a-f]{40}/i
    );
  });
});

describe("isCel2()", () => {
  test("l1", async () => {
    const web3 = new Web3(CeloChains.alfajores.rpcUrl);
    const plugin = new CeloTransactionTypesPlugin();
    web3.registerPlugin(plugin);
    expect(isCel2(plugin)).resolves.toBe(false);
  });
  test("l2", async () => {
    const web3 = new Web3(CeloChains.dango.rpcUrl);
    const plugin = new CeloTransactionTypesPlugin();
    web3.registerPlugin(plugin);
    expect(isCel2(plugin)).resolves.toBe(true);
  });
});
