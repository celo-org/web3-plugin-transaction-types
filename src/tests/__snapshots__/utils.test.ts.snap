// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`CeloChains chains should be configured 1`] = `
{
  "alfajores": {
    "chainId": 44787,
    "explorerUrl": "https://alfajores.celoscan.io",
    "rpcUrl": "https://alfajores-forno.celo-testnet.org",
  },
  "celo": {
    "chainId": 42220,
    "explorerUrl": "https://celoscan.io",
    "rpcUrl": "https://forno.celo.org",
  },
  "dango": {
    "chainId": 44787,
    "explorerUrl": "https://celo-dango.blockscout.com/",
    "rpcUrl": "https://forno.dango.celo-testnet.org",
  },
  "mainnet": {
    "chainId": 42220,
    "explorerUrl": "https://celoscan.io",
    "rpcUrl": "https://forno.celo.org",
  },
  "testnet": {
    "chainId": 44787,
    "explorerUrl": "https://alfajores.celoscan.io",
    "rpcUrl": "https://alfajores-forno.celo-testnet.org",
  },
  "testnetl2": {
    "chainId": 44787,
    "explorerUrl": "https://celo-dango.blockscout.com/",
    "rpcUrl": "https://forno.dango.celo-testnet.org",
  },
}
`;

exports[`Types tx types should be configured 1`] = `
{
  "cip42": "0x7c",
  "cip64": "0x7b",
  "cip66": "0x7a",
  "eip1559": "0x02",
  "ethereum-legacy": "",
}
`;